src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts

```diff
@@ -13,33 +13,73 @@ import virtualColumn = require("widgets/virtualGrid/columns/virtualColumn");
 
 class runQueryFeature implements columnPreviewFeature {
-    install($tooltip: JQuery, valueProvider: () => any, elementProvider: () => any, containerSelector: string) {
+    install($tooltip: JQuery, valueProvider: () => any, elementProvider: () => any, containerSelector: string): void {
         $tooltip.on("click", ".run-query", () => {
-            const value = valueProvider();
-
+            let value = valueProvider();
             const item: Raven.Client.Documents.Changes.TrafficWatchChangeBase = elementProvider();
-            
-            const query = queryCriteria.empty();
 
-            query.queryText(value);
-            query.name("Traffic watch query"); 
-            query.recentQuery(true);
+            if (item.TrafficWatchType !== "Http" || (item as TrafficWatchHttpChange).Type !== "MultiGet") {
+                runQueryFeature.executeQuery(value, item);
+                return;
+            }
+
+            const queryList = runQueryFeature.getQueryList(value);
+            if (queryList.length === 1) {
+                runQueryFeature.executeQuery(queryList[0], item);
+            } else {
+                app.showBootstrapDialog(new trafficWatchQueriesDialog(queryList))
+                    .done(queryToExecute => {
+                        if (queryToExecute) {
+                            runQueryFeature.executeQuery(queryToExecute, item);
+                        }
+                    });
+            }
+        });
+    }
 
-            const queryDto = query.toStorageDto();
+    private static getQueryList(value: any): string[] {
+        const queryList: string[] = [];
 
-            const db = databasesManager.default.getDatabaseByName(item.DatabaseName);
-            
-            recentQueriesStorage.saveAndNavigate(db, queryDto, { newWindow: true });
+        const lines = value.split('\r\n');
+        lines.forEach((line: string) => {
+            if (line) {
+                // parse & strip the "?query=" from beginning of text
+                const query = JSON.parse(line).Query.slice(7);
+                if (query) {
+                    queryList.push(query);
+                }
+            }
         });
+
+        return queryList;
+    }
+    
+    private static executeQuery(value: any, item: Raven.Client.Documents.Changes.TrafficWatchChangeBase): void {
+        const query = queryCriteria.empty();
+
+        query.queryText(value);
+        query.name("Traffic watch query");
+        query.recentQuery(true);
+
+        const queryDto = query.toStorageDto();
+
+        const db = databasesManager.default.getDatabaseByName(item.DatabaseName);
+
+        recentQueriesStorage.saveAndNavigate(db, queryDto, { newWindow: true });
     }
     
-    syntax(column: virtualColumn, escapedValue: any) {
+    syntax(column: virtualColumn, escapedValue: any, element: any): string {
+        const buttonText = element.Type === 'MultiGet' ? "RunQuery..." : "RunQuery";
```

Comment:
we might lie a bit here but I think this is good approximation ;-)

Reply:
Refactored this method so that 'RunQuery' button only shows when relevant